tools/data_utils.py

```diff
@@ -85,86 +85,109 @@ def processImage(img, imWidth, imHeight, num_patches):
     return img
 
 
-def get_filtered_name_paths(filtered_names_path):
+def get_filtered_imageNames(filtered_imageNames_path):
 
-    assert os.path.isfile(filtered_names_path), "The file path {} is not a valid".format(filtered_names_path)
+    assert os.path.isfile(filtered_imageNames_path), "The file path {} is not a valid".format(filtered_imageNames_path)
 
-    with open(filtered_names_path) as f:
+    with open(filtered_imageNames_path) as f:
         content = f.read().splitlines()
-        
-    filtered_index = [int(''.join(x for x in char if x.isdigit())) for char in content] 
+    
+    return content
 
-    return filtered_index
 
+def filter_frames(frames, labels, paths_used, type, shuffled_indices, imgLists, offset_after_skip, num_labels, num_test_labels):
+    
+    # shuffle data 
+    paths_used[:len(shuffled_indices)] = paths_used[shuffled_indices]
+    frames[:len(shuffled_indices)] = frames[shuffled_indices]
+    
+    if len(imgLists) == 2:
```

Comment:
What is this special case about?

Reply:
Updated function to enabled shuffling of N traverses as opposed to just 2 for the reference dataset.